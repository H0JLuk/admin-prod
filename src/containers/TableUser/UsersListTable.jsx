import React, { useMemo } from 'react';
import cn from 'classnames';
import { Pagination, Select, Table } from 'antd';

import styles from './UsersListTable.module.css';

const USER_ACTIVE = 'Активен';
const USER_BLOCKED = 'Заблокирован';
const LOGIN = 'Логин';
const LOCATION = 'Локация';
const SALE_POINT = 'Точка продажи';
const PAGE_SIZES = [10, 20, 50, 100];

const UsersListTable = ({
    loadingData,
    onRow,
    rowSelection,
    dataSource,
    pagination,
    onChangePage,
    onChangePageSize,
}) => {
    const pageSizes = useMemo(() => {
        if (!PAGE_SIZES.includes(Number(pagination.pageSize))) {
            return [...PAGE_SIZES, pagination.pageSize].sort((a, b) => a - b);
        }

        return PAGE_SIZES;
    }, [pagination.pageSize]);

    return (
        <div className={ styles.tableWrapper }>
            <Table
                loading={ loadingData }
                onRow={ onRow }
                rowKey="id"
                className={ styles.table }
                rowSelection={ rowSelection }
                dataSource={ dataSource }
                pagination={ false }
                showHeader={ false }
            >
                <Table.Column
                    width="25%"
                    key="login"
                    dataIndex="personalNumber"
                    render={ (item) => columnRender(item, LOGIN) }
                />
                <Table.Column
                    width="35%"
                    key="locationName"
                    dataIndex="locationName"
                    render={ (item) => columnRender(item, LOCATION) }
                />
                <Table.Column
                    width="25%"
                    key="salePointName"
                    dataIndex="salePointName"
                    render={ (item) => columnRender(item, SALE_POINT) }
                />
                <Table.Column
                    width="15%"
                    key="status"
                    dataIndex="tmpBlocked"
                    className={ styles.cellRight }
                    render={ columnRenderStatus }
                />
            </Table>
            <div className={ styles.paginationWrapper }>
                <Pagination
                    { ...pagination }
                    className={ styles.pagination }
                    onChange={ onChangePage }
                />
                <Select
                    className={ styles.pageSizes }
                    defaultValue={ PAGE_SIZES[0] }
                    onChange={ onChangePageSize }
                    value={ pagination.pageSize }
                >
                    {pageSizes.map((size) => (
                        <Select.Option key={ size }>{ size }</Select.Option>
                    ))}
                </Select>
            </div>
        </div>
    );
};

export default UsersListTable;


const columnRender = (item, name) => (
    <>
        <div className={ styles.tableTitle }>{ name }</div>
        <div className={ styles.tableCell }>{ item }</div>
    </>
);

const columnRenderStatus = (blocked) => (
    <div className={ cn({
        [styles.blocked]: blocked,
        [styles.active]: !blocked
    }) }>
        { blocked ? USER_BLOCKED : USER_ACTIVE }
    </div>
);
