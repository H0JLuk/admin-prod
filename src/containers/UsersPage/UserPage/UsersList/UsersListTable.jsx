import React from 'react';
import cn from 'classnames';
import { Empty, Table } from 'antd';
import { ROLES_RU } from '../../../../constants/roles';
import { LOGIN_TYPE } from '../../../../constants/loginTypes';

import styles from './UsersListTable.module.css';

const USER_ACTIVE = 'Активен';
const USER_BLOCKED = 'Заблокирован';
const LOGIN = 'Логин';
const LOCATION = 'Локация';
const SALE_POINT = 'Точка продажи';
const ROLE = 'Роль';
const LOGIN_METHOD = 'Способ входа';
const EMPTY_TABLE = {
    firstMessagePart: 'Мы ничего не нашли.',
    secondMessagePart: 'Измените значение поиска и попробуйте еще раз',
};

const EmptyMessage = (
    <Empty description={ null } >
        <div>{ EMPTY_TABLE.firstMessagePart }</div>
        <div>{ EMPTY_TABLE.secondMessagePart }</div>
    </Empty>
);

const UsersListTable = ({
    loadingData,
    onRow,
    rowSelection,
    dataSource,
    pagination,
    onChangePage,
}) => (
    <div className={ styles.tableWrapper }>
        <Table
            loading={ loadingData }
            onRow={ onRow }
            rowKey="id"
            className={ styles.table }
            rowSelection={ rowSelection }
            dataSource={ dataSource }
            pagination={ pagination }
            showHeader={ false }
            onChange={ onChangePage }
            locale={ { emptyText: EmptyMessage } }
        >
            <Table.Column
                width="20%"
                key="login"
                dataIndex="personalNumber"
                render={ (item) => columnRender(item, LOGIN) }
            />
            <Table.Column
                width="20%"
                key="locationName"
                dataIndex="locationName"
                render={ (item) => columnRender(item, LOCATION) }
            />
            <Table.Column
                width="20%"
                key="salePointName"
                dataIndex="salePointName"
                render={ (item) => columnRender(item, SALE_POINT) }
            />
            <Table.Column
                width="25%"
                key="loginType"
                dataIndex="loginType"
                render={ (item) => columnRender(LOGIN_TYPE[item], LOGIN_METHOD) }
            />
            <Table.Column
                width="15%"
                key="role"
                dataIndex="role"
                render={ (userRole) => columnRender(ROLES_RU[userRole], ROLE) }
            />
            <Table.Column
                width="15%"
                key="status"
                dataIndex="tmpBlocked"
                className={ styles.cellRight }
                render={ columnRenderStatus }
            />
        </Table>
    </div>
);

export default UsersListTable;


const columnRender = (item, name) => (
    <>
        <div className={ styles.tableTitle }>{ name }</div>
        <div className={ styles.tableCell }>{ item }</div>
    </>
);

const columnRenderStatus = (blocked) => (
    <div className={ cn({
        [styles.blocked]: blocked,
        [styles.active]: !blocked
    }) }>
        { blocked ? USER_BLOCKED : USER_ACTIVE }
    </div>
);
