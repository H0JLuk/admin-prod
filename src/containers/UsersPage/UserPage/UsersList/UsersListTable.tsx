import React from 'react';
import cn from 'classnames';
import { Empty, Table, TableProps } from 'antd';
import { UserInfo } from '@types';
import ROLES, { ROLES_RU } from '@constants/roles';
import { LOGIN_TYPE, LoginTypes } from '@constants/loginTypes';

import styles from './UsersListTable.module.css';

type UsersListTableProps = {
    loadingData: boolean;
    onRow: TableProps<UserInfo>['onRow'];
    rowSelection?: TableProps<UserInfo>['rowSelection'];
    dataSource: UserInfo[];
    pagination: TableProps<UserInfo>['pagination'];
    onChangePage: TableProps<UserInfo>['onChange'];
};

const USER_ACTIVE = 'Активен';
const USER_BLOCKED = 'Заблокирован';
const LOGIN = 'Логин';
const LOCATION = 'Локация';
const SALE_POINT = 'Точка продажи';
const ROLE = 'Роль';
const LOGIN_METHOD = 'Способ входа';
const PARTNER = 'Партнёр';
const EMPTY_TABLE = {
    firstMessagePart: 'Мы ничего не нашли.',
    secondMessagePart: 'Измените значение поиска и попробуйте еще раз',
};

const EmptyMessage = (
    <Empty description={null} >
        <div>{EMPTY_TABLE.firstMessagePart}</div>
        <div>{EMPTY_TABLE.secondMessagePart}</div>
    </Empty>
);

const UsersListTable: React.FC<UsersListTableProps> = ({
    loadingData,
    onRow,
    rowSelection,
    dataSource,
    pagination,
    onChangePage,
}) => (
    <div className={styles.tableWrapper}>
        <Table<UserInfo>
            loading={loadingData}
            onRow={onRow}
            rowKey="id"
            className={styles.table}
            rowSelection={rowSelection}
            dataSource={dataSource}
            pagination={pagination}
            showHeader={false}
            onChange={onChangePage}
            locale={{ emptyText: EmptyMessage }}
        >
            <Table.Column
                width="15%"
                key="login"
                dataIndex="personalNumber"
                render={(item) => columnRender(item, LOGIN)}
            />
            <Table.Column
                width="15%"
                key="locationName"
                dataIndex="locationName"
                render={(item) => columnRender(item, LOCATION)}
            />
            <Table.Column
                width="15%"
                key="salePointName"
                dataIndex="salePointName"
                render={(item) => columnRender(item, SALE_POINT)}
            />
            <Table.Column
                width="15%"
                key="parentUserName"
                dataIndex="parentUserName"
                render={(item) => columnRender(item || <i>Нет партнёра</i>, PARTNER)}
            />
            <Table.Column
                width="15%"
                key="loginType"
                dataIndex="loginType"
                render={(item: LoginTypes) => columnRender(LOGIN_TYPE[item], LOGIN_METHOD)}
            />
            <Table.Column
                width="10%"
                key="role"
                dataIndex="role"
                render={(userRole: ROLES) => columnRender(ROLES_RU[userRole], ROLE)}
            />
            <Table.Column
                width="15%"
                key="status"
                dataIndex="tmpBlocked"
                className={styles.cellRight}
                render={columnRenderStatus}
            />
        </Table>
    </div>
);

export default UsersListTable;


const columnRender = (item: string, name: string) => (
    <>
        <div className={styles.tableTitle}>{name}</div>
        <div className={styles.tableCell}>{item}</div>
    </>
);

const columnRenderStatus = (blocked: UserInfo['tmpBlocked']) => (
    <div className={cn({
        [styles.blocked]: blocked,
        [styles.active]: !blocked,
    })}>
        {blocked ? USER_BLOCKED : USER_ACTIVE}
    </div>
);
