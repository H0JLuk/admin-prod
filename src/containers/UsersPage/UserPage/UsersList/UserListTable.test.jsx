import React from 'react';
import { shallow } from 'enzyme';
import UsersListTable from './UsersListTable';

describe('UserListTable test', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<UsersListTable />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
