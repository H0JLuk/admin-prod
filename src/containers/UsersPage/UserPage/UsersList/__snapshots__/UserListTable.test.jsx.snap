// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListTable test should match snapshot 1`] = `
"<div className=\\"tableWrapper\\">
  <Table loading={[undefined]} onRow={[undefined]} rowKey=\\"id\\" className=\\"table\\" rowSelection={[undefined]} dataSource={[undefined]} pagination={[undefined]} showHeader={false} onChange={[undefined]} locale={{...}}>
    <Column width=\\"20%\\" dataIndex=\\"personalNumber\\" render={[Function: render]} />
    <Column width=\\"20%\\" dataIndex=\\"locationName\\" render={[Function: render]} />
    <Column width=\\"20%\\" dataIndex=\\"salePointName\\" render={[Function: render]} />
    <Column width=\\"25%\\" dataIndex=\\"loginType\\" render={[Function: render]} />
    <Column width=\\"15%\\" dataIndex=\\"role\\" render={[Function: render]} />
    <Column width=\\"15%\\" dataIndex=\\"tmpBlocked\\" className=\\"cellRight\\" render={[Function: columnRenderStatus]} />
  </Table>
</div>"
`;
